(Docs) Added type = "file" reference `[tool.poetry.scripts]` (#9572)
python-poetry/poetry: docs/pyproject.md

When a script is added or updated, run `poetry install` to make them available in the project's virtualenv.
{{% /note %}}

```toml
[tool.poetry.scripts]
my_executable = { reference = "some_binary.exe", type = "file" }
```

With this configuration, the file `some_binary.exe` is looked for in the directory of the active `pyproject.toml` and copied into the installation folder.

> On Windows, the `.exe` extension is added to the file and the file will be placed in the `Scripts/` directory.
> On Linux no extension is added and the destination is `bin/`.

Reviewer: ```suggestion
* On Windows, the `.exe` extension is added to the file, and is placed in the `Scripts/` directory.
* On Linux no extension is added, and is placed in the `bin/` directory.
```

For better readability, perhaps?

Reviewer 2: This is ungrammatical.  "The '.exe' extension... is placed in the 'Scripts/' directory" is not what you intend.

Reviewer 3: I agree the readability of this is not optimal. From the way it is currently described the full paragraph seems to imply that on Windows you'd end up with `some_file.exe.exe`

> With this configuration, the file `some_binary.exe` is looked for in the directory....On Windows, the `.exe` extension is added to the file

Perhaps something like this
> Poetry will look for the referenced file in the directory of the active project and then copied to the OS installation directory.
> * On Windows the `.exe` extension is added to the file and it is copied to the Python Scripts directory
> * On *nix systems the file will be copied to the environment `bin` directory

Author: Updates the text. The `.exe` bit was weird indeed, it should read clearly now.

When a script is added or updated, run `poetry install` to make them available in the project's virtualenv.
{{% /note %}}

> ```toml

Reviewer 4: Can you also update the `deprecation` notice above to include something like "This section should only be used for scripts of type 'file'".

Author: I've added that text.
But I don't actually understand what that means, has the `scripts` section been deprecated entirely?

Reviewer 4: Might need to reword it if it is not clear I guess :) 

PEP 621 introduced both cli and gui scripts in the spec, but lacks the `type = "file"` functionality. So, the prefered way of doing cli/gui scripts is the `project.scripts` section.

When a script is added or updated, run `poetry install` to make them available in the project's virtualenv.
{{% /note %}}

```toml
[tool.poetry.scripts]
my_executable = { reference = "some_binary.exe", type = "file" }
```

> With this configuration, Poetry will look for the referenced file in the active project and then copy it to the OS installation directory.

Reviewer 4: ```suggestion
This tells Poetry to include the specified file, relative to your project directory, in distribution builds. It will then be copied to the appropriate installation directory for your operating system when your package is installed. 
```

* On Windows the file is placed in the `Scripts/` directory.
* On *nix system the file is placed in the `bin/` directory.

> ## `extras`

Reviewer 4: ```suggestion
### `extras`
```
